salt/returners/mysql.py

```diff
@@ -200,9 +200,15 @@ def _get_options(ret=None):
                                                    __salt__=__salt__,
                                                    __opts__=__opts__,
                                                    defaults=defaults)
-    # Ensure port is an int
-    if 'port' in _options:
-        _options['port'] = int(_options['port'])
+    # post processing
+    for k, v in _options.iteritems(): 
+        if isinstance(v, basestring) and v.lower() == 'none':
```

Comment:
We may want this to be `six.string_types` so we have forward and backward compatability.